core/src/main/scala/org/apache/spark/SparkEnv.scala

```diff
@@ -90,17 +90,25 @@ class SparkEnv (
   private var driverTmpDirToDelete: Option[String] = None
 
   private[spark] def stop() {
+
+    if(isStopped) return
```

Comment:
Heh, I would have gone with just 

```
if (isStopped) {
  return
}
```

to avoid the extra running indent but leave it.